pkg/utils/symbol_table.go

```diff
@@ -0,0 +1,212 @@
+package utils
+
+import (
+	"errors"
+	"sort"
+	"sync"
+)
+
+// The Symbol interface defines what is needed from a symbol implementation in
+// order to facilitate the lookup functionalities provided by SymbolTable.
+// Implementations of Symbol can hold various types of information relevant to
+// the type of symbol they represent.
+type Symbol interface {
+	Name() string
+	Address() uint64
```

Comment:
Considering the test implementation this seems to be more along the lines of a base address right? As `Contains` should return whether some address is inside an address range defined by the symbol's implemenetation.

Could you maybe add some fast one-line docs to each interface method?